hashintel/hash
# Refactor `me` GQL query to return a sub-graph, deprecate `accounts` GQL query

## 🌟 What is the purpose of this PR?




As part of the work for [deprecating the `User` GQL type definition](https://app.asana.com/0/1203007126736610/1203244611385365/f), this PR:
- replaces the return type of the `me` GQL query with a sub-graph
- deprecates the `accounts` GQL query

## 🔗 Related links





- [Asana task](https://app.asana.com/0/0/1203251067111510/f) _(internal)_

## 🔍 What does this change?




- introduces user TS types on the frontend (`MinimalUser`, `User` and `AuthenticatedUser`)
- introduces org TS types on the frontend (`MinimalOrg` and `Org`)
- introduces mapping functions to map from a sub-graph to its corresponding user/org frontend TS type
- replaces the `useUser` hook with a refactored `useAuthenticatedUser` hook
- fixes cascading lint errors

## 📜 Does this require a change to the docs?



Not that I'm aware of.

## ⚠️ Known issues



## 🐾 Next steps



## 🛡 What tests cover this?



None, as frontend tests aren't currently implemented.

## ❓ How to test this?



1. Run the dev worskspace
2. Login, and navigate to a workspace page to verify that getting the current user/its orgs is working.

## 📹 Demo



https://user-images.githubusercontent.com/42802102/198535477-25a26c49-6585-47f0-bb46-3d8524fac879.mov
